MetaMask/metamask-extension
# feat: Add DappPermissionModal

## **Description**

- Implements the DappPermissionModal
- Introduces a small change to Permission Cell as to allow not showing an information button on the side
- Connects the Modal so it can open and close
- Implements the disconnect button


[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/23237?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2166

## **Manual testing steps**

1. Open the Connected Accounts panel
2. Select an Account and click on Permission details
3. Check that the permissions are properly rendered, and that the user name is properly set
4. Disconnect the account

## **Screenshots/Recordings**

<img width="339" alt="Screenshot 2024-03-12 at 04 12 28" src="https://github.com/MetaMask/metamask-extension/assets/11148144/bd9071ab-dac1-4509-8fac-d58f6d6698f9">


### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
